docs/content/docs/external_auth_opa.md

```diff
@@ -0,0 +1,119 @@
+
+## Demo Walkthrough
+
+- Deploy an `opa-envoy-plugin`, use curated yaml in this folder.
+  - NOTE: This POC is using a single plugin to handle all pod connections, and through the network. This is not intended for production.
+```
+kubectl create namespace opa
+kubectl apply -f docs/example/manifests/opa/deploy-opa-envoy.yaml
+```
+`deploy-opa-envoy.yaml` will deploy OPA's envoy plugin and run it as a service, allowing external authorization calls from envoys through the network.
+
+- Deploy OSM's Demo, follow `demo/run-osm-demo.sh`
+```
+demo/run-osm-demo.sh  # wait for all services to come up
+```
+- This branch has an External Authorization server expected by default at `opa.opa.svc.cluster.local:9191`. Timeout for authorization RTT is `1s` by default, and will not allow traffic in case of failure.
+
+Traffic should fail right out of the bat:
+```
+kubectl logs <bookbuyer_pod> -n bookbuyer bookbuyer
+```
+```
+...
+--- bookbuyer:[ 8 ] -----------------------------------------
+
+Fetching http://bookstore.bookstore:14001/books-bought
+Request Headers: map[Client-App:[bookbuyer] User-Agent:[Go-http-client/1.1]]
+Identity: n/a
+Booksbought: n/a
+Server: envoy
+Date: Tue, 04 May 2021 01:20:39 GMT
+Status: 403 Forbidden
+ERROR: response code for "http://bookstore.bookstore:14001/books-bought" is 403;  expected 200
+...
+```
+
+You shoud also be able to see the logs in `opa-envoy-plugin` for rejected authorization calls:
+```
+kubectl logs <opa_pod> -n opa
+```
+```
+{"decision_id":"1df154b5-658a-47bf-ac18-be52998605da","input":{"attributes":{"destination":{"address":{"socketAddress":{"address":"10.0.16.44","portValue":14001}}},"metadataContext":{},"request":{"http":{"headers":{":authority":"bookstore.bookstore:14001",":method":"GET",":path":"/books-bought","accept-encoding":"gzip","client-app":"bookbuyer","user-agent":"Go-http-client/1.1","x-forwarded-proto":"http","x-request-id":"69b80716-6af4-4986-bf8c-8f209d96f131"},"host":"bookstore.bookstore:14001","id":"6079090369556950701","method":"GET","path":"/books-bought","protocol":"HTTP/1.1"},"time":"2021-05-04T01:21:18.195876Z"},"source":{"address":{"socketAddress":{"address":"10.244.2.10","portValue":53488}}}},"parsed_body":null,"parsed_path":["books-bought"],"parsed_query":{},"truncated_body":false,"version":{"encoding":"protojson","ext_authz":"v3"}},"labels":{"id":"6e56bc11-a212-4c3e-be4d-b33186fd581d","version":"0.28.0-envoy"},"level":"info","metrics":{"timer_rego_query_eval_ns":105799,"timer_server_handler_ns":425097},"msg":"Decision Log","path":"envoy/authz/allow","requested_by":"","result":false,"time":"2021-05-04T01:21:18Z","timestamp":"2021-05-04T01:21:18.1971808Z","type":"openpolicyagent.org/decision_logs"}
+```
+
+- Now edit OPA's policy:
+```
+kubectl edit configmap opa-policy -n opa
+```
+change specifically the default-all from:
+```
+default allow = false
+```
+to
+```
+default allow = true
```

Comment:
This seems to be updating the default. Is there an ask to support more granular policies?

Reply:
there is, but it's a matter of writting the polici down in Rego format. For the sake of simplicity I assumed just tweaking a default allow/do-not-allow var

Reply:
ref: https://www.openpolicyagent.org/docs/latest/policy-language/

Reply:
Should we drop one more example for a more granular rule?

Reply:
I'll might augment that on `main`, but I plan to leave it like this in this branch for now